nifi-api/src/main/java/org/apache/nifi/components/resource/Utf8TextResource.java

```diff
@@ -46,6 +47,11 @@ public InputStream read() throws IOException {
         return new ByteArrayInputStream(text.getBytes(StandardCharsets.UTF_8));
     }
 
+    @Override
+    public InputStreamReader createReader() throws IOException {
+        return new InputStreamReader(this.read(), StandardCharsets.UTF_8);
+    }
```

Comment:
Although this is convenient, it is best to avoid introducing it as an API-level method.